operators/ior_presets.py

```diff
@@ -0,0 +1,181 @@
+            preset_list.append((item[0], text, text, item[1][2]))
+        return preset_list
+
+    @classmethod
+    def get_item(cls, key=default_key):
+        ret = None
+        if key not in cls._ior_values.keys():
+            print("Error: key '{}' not found in "
+                  "LuxCoreIORPresets::get_item()".format(key))
+        else:
+            ret = cls._ior_values[key]
+        return ret
+
+    @classmethod
+    def get_value(cls, key=default_key):
+        ret = None
+        if key not in cls._ior_values.keys():
+            print("Error: key '{}' not found in "
+                  "LuxCoreIORPresets::get_value()".format(key))
+        else:
+            ret = cls._ior_values[key][1]
+        return ret
+
+
+# It seems that Blender *Property classes do not support inheritance if
+# there are methods in the same class. The following classes
+# (LuxCoreIORPresetCommonProperties, LuxCoreIORPresetBase,
+#  LUXCORE_OT_ior_preset_names  and LUXCORE_OT_ior_preset_values) are an
+# attempt at minimizing code duplication within the limits imposed by Blender.
+class LuxCoreIORPresetCommonProperties():
+    """ A property-only class to work around Blender's inheritance limits. """
+    bl_idname = "luxcore.ior_preset_common_properties"
+    bl_label = ""
+    node_name = StringProperty()
+    node_tree_name = StringProperty()
+
+
+class LuxCoreIORPresetBase(bpy.types.Operator,
+                           LuxCoreIORPresetCommonProperties):
+    """ A base class for LUXCORE_OT_ior_preset_* common methods. Do not call"""
+
+    bl_idname = "luxcore.ior_preset_base"
+
+    def cb_ior_preset(self, context):
+        return []
+
+    ior_preset = EnumProperty(name="IOR Preset",
+                              description="Index of Refraction Preset Values",
+                              items=cb_ior_preset)
+
+    def execute(self, context):
+        ior = LuxCoreIORPresetValues.get_item(self.ior_preset)
+        node_data_path = bpy.data.node_groups[self.node_tree_name].\
+            nodes[self.node_name]
+        node_data_path.ior_key = self.ior_preset
+        node_data_path.ior_name_text = ior[0]
+        node_data_path.ior_value_text = str(ior[1])
+        node_data_path.ior_value_float = ior[1]
+        return {'FINISHED'}
+
+    def invoke(self, context, event):
+        context.window_manager.invoke_search_popup(self)
+        return {'FINISHED'}
+
+
+class LUXCORE_OT_ior_preset_names(LuxCoreIORPresetBase,
+                                  LuxCoreIORPresetCommonProperties):
+    """ A custom operator to return a list of IOR presets sorted by name """
+
+    bl_idname = "luxcore.ior_preset_names"
+    bl_description = "Index of Refraction presets sorted by name"
+    bl_property = "ior_preset"
+    node_name = StringProperty()
+    node_tree_name = StringProperty()
+
+    def cb_ior_preset(self, context):
+        preset_list = []
+        cl_presets = LuxCoreIORPresetValues()
```

Comment:
Isn't `LuxCoreIORPresetValues` supposed to be used as a singleton? Why do you make an instance here?
You can make `get_sorted_list` into a classmethod.
(same below in the other operator)